src/Dispatcher.cpp

```diff
@@ -243,7 +248,7 @@ void DispatcherBase::DispatchThreadMain() {
     if (!m_active) break;  // in case we were woken up to terminate
 
     // perform periodic persistent save
-    if (m_server && !m_persist_filename.empty() && start > next_save_time) {
+    if (m_networkMode & NT_NET_MODE_SERVER && !m_persist_filename.empty() && start > next_save_time) {
```

Comment:
I prefer `(m_networkMode & NT_NET_MODE_SERVER) != 0` to avoid any potential order-of-operations confusion.